frontend/src/components/common/EditorDialog.tsx

```diff
@@ -293,15 +293,26 @@ function DocsViewer(props: {docSpecs: any}) {
   const { docSpecs } = props;
   const classes = useStyles();
   const [docs, setDocs] = React.useState<object | null>(null);
-
+  const [docsError, setDocsError] = React.useState<string | null>(null);
   React.useEffect(() => {
+    if (docSpecs.error) {
+      setDocsError(`Cannot load documentation: ${docSpecs.error}`);
+      return;
+    }
     if (!docSpecs.apiVersion || !docSpecs.kind) {
+      setDocsError(
+        `Cannot load documentation: 
+        Please make sure the YAML is valid and has the kind and apiVersion set.
+        `);
```

Comment:
Nitpick: Are you using the backquotes here for preserving the space or was it a mistake? You can use a longer than 100 chars line if it's a string (with single quotes).